src/Compilers/CSharp/Portable/Compiler/TypeCompilationState.cs

```diff
@@ -164,6 +172,59 @@ public void Free()
             }
 
             _wrappers = null;
+            _constructorInitializers = null;
+        }
+
+        // Add an edge to the ctor-initializer graph
+        internal void NoteConstructorInitializer(MethodSymbol method1, MethodSymbol method2, CSharpSyntaxNode syntax)
+        {
+            if (_constructorInitializers == null)
+            {
+                _constructorInitializers = new Dictionary<MethodSymbol, Tuple<MethodSymbol, CSharpSyntaxNode>>();
```

Comment:
The implementation reports the one error on the (lexically) last constructor in each distinct ctor-initializer cycle. It does not depend on the order of enumeration of the keys anywhere.